src/js/components/Anchor.js

```diff
@@ -3,14 +3,27 @@
 import React, { Children, PropTypes } from 'react';
 import classnames from 'classnames';
 import RightLeftIcon from './icons/LinkRightLeft';
+import iconsMap from '../index-icons';
 
 const CLASS_ROOT = 'anchor';
 
 const Anchor = props => {
   let icon;
 
-  if (props.primary) {
-    icon = <RightLeftIcon />;
+  if (props.icon && !iconsMap.hasOwnProperty(props.icon)) {
```

Comment:
The logic seems a bit convoluted here. We are testing props.icon and whether we have the icon multiple times. How about:

if (props.icon) {
  let CustomIcon = iconsMap[props.icon];
  if (! CustomIcon) {
    console.warn();
  } else {
    icon = <CustomIcon />;
  }
} else if (props.primary) {
  icon = <RightLeftIcon />;
}
if (icon) {
  icon = <span ...>{icon}</span>;
}